### ISSUE TITLE
ParamSpec with Generic leads to false positive 

### ISSUE THREAD

**Issue by vivax3794** [microsoft/pyright#2705]
**Describe the bug**
with the following code
```python
from typing import ParamSpec, Concatenate, Generic, Callable, Any

P = ParamSpec("P")

class Foo(Generic[P]):
    def __init__(self, func: Callable[P, Any]) -> None: ...
def bar(baz: Foo[Concatenate[int, P]]) -> Foo[P]: ...

def test(a: int, b: str) -> str: ...

abc = Foo(test)
reveal_type(abc)
bar(abc)
```
pyright produces the following output:
```
/home/vivax/coding/VivCord/test.py
  /home/vivax/coding/VivCord/test.py:12:13 - info: Type of "abc" is "Foo[(a: int, b: str)]"
  /home/vivax/coding/VivCord/test.py:13:5 - error: Argument of type "Foo[(a: int, b: str)]" cannot be assigned to parameter "baz" of type "Foo[(Concatenate[int, P@bar])]" in function "bar"
    TypeVar "P@Foo" is invariant
      Type "(a: int, b: str)" cannot be assigned to type "(Concatenate[int, P@bar])"
        Parameter 1: type "Concatenate[int, P@bar]" cannot be assigned to type "int"
        Function accepts too few positional parameters; expected 2 but received 1
        Keyword parameter "b" is missing in destination (reportGeneralTypeIssues)
```

**Expected behavior**
from what I can tell there is nothing wrong with the code and I would excpect the result of the call to be `Foo[(b: str)])`

**VS Code extension or command-line**
Command-line
**version:** pyright 1.1.195
```
[tool.pyright]
typeCheckingMode = "strict"
reportPropertyTypeMismatch = "error"
reportUninitializedInstanceVariable = "error"
reportCallInDefaultInitializer = "error"
reportImplicitStringConcatenation = "error"
reportImportCycles = "none"
```

**Comment by erictraut** [comment]
Thanks for reporting this. There was a bug that affected the explicit specialization of a generic class that used `Concatenate` in the type argument, as in `Foo[Concatenate[int, P]]`. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.196, which I just published. It will also be included in the next release of pylance.
